framework/encode/api_capture_manager.h

```diff
@@ -47,11 +47,11 @@ class ApiCaptureManager
     virtual void DestroyStateTracker()                                 = 0;
     virtual void WriteTrackedState(util::FileOutputStream* file_stream,
                                    format::ThreadId        thread_id,
-                                   util::FileOutputStream* asset_file_stream,
-                                   const std::string&      asset_file_name) = 0;
+                                   util::FileOutputStream* asset_file_stream = nullptr,
```

Comment:
I would advice against default values in virtual functions. An alternative which I like more would be to introduce a new function. E.g.:

```cpp
virtual void WriteTrackedState(util::FileOutputStream* file_stream, format::ThreadId thread_id) = 0;
virtual void WriteTrackedStateWithAsset(util::FileOutputStream* file_stream,
                                   format::ThreadId        thread_id,
                                   util::FileOutputStream* asset_file_stream,
                                   const std::string&      asset_file_name) = 0;
```

Reply:
Actually this is a very good idea. Thanks, Antonio